ocaml/xapi/message_forwarding.ml

```diff
@@ -3387,14 +3387,14 @@ functor
           (host_uuid ~__context self) ;
         Local.Host.get_vms_which_prevent_evacuation ~__context ~self
 
-      let evacuate ~__context ~host ~network =
+      let evacuate ~__context ~host ~network ?evacuate_batch_size =
```

Comment:
It's not a good idea to add optional arguments to a function without non-labelled arguments, it can lead to partially-applied functions. Unfortunately, a unit parameter can't just be added here.

I recommend making it a labelled argument, the end result is the same: users have to define whether they want to set a custom value of the parameter with an `Option` and this has to get treated in the function body, but the possible errors are avoided.

Note that on the API side it's still defined as an optional argument, this is all about how it's treated / represented on the ocaml side

Reply:
When using an `Option` I get `Fatal error: exception Failure("option types never stored in the database")`

So i think I'll give int64 and ignore when it 0L

Reply:
My point was to _not_ change the IDL, it was fine as it was

Reply:
In `datamodel_host.ml`, the new parameter is defined as `param_type= Int`.

Reply:
It seems this kind of `versioned_params` is always  of `Optional` to support backward compatibility?

Reply:
I'm not sure to understand your last comment.
It means in the new arg is optionnal only in the API but is not in the implementation?

Reply:
Yes. That's what I read from the code. A quick experiment could make it confirmed.